chapters/functions.tex

```diff
@@ -2331,65 +2333,110 @@ \subsection{Aliasing}\label{aliasing}
 
 \subsection{Annotations for External Libraries and Include Files}\label{annotations-for-external-libraries-and-include-files}
 
-The following annotations are useful in the context of calling external functions from Modelica, and they should occur on the \lstinline!external!-clause and no other standard annotations should occur on the \lstinline!external!-clause.
-They can all specify either a scalar value or an array of values as indicated below for the \lstinline!Library! annotation:
+The annotations listed below are useful in the context of calling external functions from Modelica, and they should occur on the \lstinline!external!-clause and no other standard annotations should occur on the \lstinline!external!-clause.
+\begin{center}
+\begin{tabular}{l|l l}
+\hline
+\tablehead{Annotation} & \tablehead{Description} & \tablehead{Details}\\
+\hline
+\hline
+\lstinline!Library! & Libraries to link with & \Cref{modelica:Library}\\
+\lstinline!Include! & Include directives & \Cref{modelica:Include}\\
+\lstinline!LibraryDirectory! & Directories containing libraries & \Cref{modelica:LibraryDirectory}\\
+\lstinline!IncludeDirectory! & Directories containing header files & \Cref{modelica:IncludeDirectory}\\
+\lstinline!SourceDirectory! & Directories containing source code& \Cref{modelica:SourceDirectory}\\
+\hline
+\end{tabular}
+\end{center}
+
+They can all specify either a scalar value or an array of values as indicated by the \lstinline[language=grammar]!string-or-string-array! syntax introduced with the \lstinline!Library! annotation.
+
+The \lstinline!ModelicaLibraryName! used below for \lstinline!IncludeDirectory!, \lstinline!LibraryDirectory!, and \lstinline!SourceDirectory! indicates the top-level class where the annotation is found in the Modelica source code.
+
+\begin{annotationdefinition}[Library]
```

Comment:
I would prefer if we just write that it should be a string or an array of strings, and don't use grammar fragments unless needed.

Reply:
Are you sure this isn't a good place to use it?  It is defined once, used in the definitions of 5 different annotations so far, and also mentioned in the running text.  It would take considerably more text and hand waving to do it without giving a name to the idea.

While we might be able to change many of the annotations to the pseudo-code style of presentation, the ones we have here belong to the category where specification using grammar actually makes sense as we can't express the type of the annotation using a Modelica type.

Reply:
Couldn't we just keep the grammar fragment for now, and revisit this once we take the bigger grip on presentation of annotations as decided at the last phone meeting (can't find in what issue that decision was noted)?

Reply:
Any chance we can return to this once we turn to the question of how to present the things that can't be expressed easily in the record style established so far?